src/dict.c

```diff
@@ -1492,10 +1469,19 @@ static void _dictShrinkIfNeeded(dict *d)
         (dict_can_resize != DICT_RESIZE_FORBID &&
          d->ht_used[0] * HASHTABLE_MIN_FILL * dict_force_resize_ratio <= DICTHT_SIZE(d->ht_size_exp[0])))
     {
-        if (!dictTypeResizeAllowed(d, d->ht_used[0]))
-            return;
-        dictShrink(d, d->ht_used[0]);
+        if (dictTypeResizeAllowed(d, d->ht_used[0]))
+            dictShrink(d, d->ht_used[0]);
+        return DICT_OK;
     }
+    return DICT_ERR;
+}
+
+static void _dictShrinkIfNeeded(dict *d) 
+{
+    /* Automatic resizing is disallowed. Return */
+    if (d->pauseAutoResize > 0) return;
```

Comment:
i'm not sure we need to move these to a separate (internal) function.
considering where dictPauseAutoResize is used (temporary disable in a narrow scope)

Reply:
Well ... I prefer to separate auto resize and "manual" resize apart, seems more reasonable.